pages/instructorGradebook/instructorGradebook.ejs

```diff
@@ -2,42 +2,57 @@
 <html lang="en">
   <head>
     <%- include('../partials/head'); %>
+    <script src="<%= node_modules_asset_path('bootstrap-table/dist/bootstrap-table.min.js') %>"></script>
+    <script src="<%= node_modules_asset_path('bootstrap-table/dist/extensions/sticky-header/bootstrap-table-sticky-header.min.js') %>"></script>
+    <link href="<%= node_modules_asset_path('bootstrap-table/dist/bootstrap-table.min.css') %>" rel="stylesheet" />
+    <link href="<%= node_modules_asset_path('bootstrap-table/dist/extensions/sticky-header/bootstrap-table-sticky-header.min.css') %>" rel="stylesheet" />
   </head>
   <script>
     $(function() {
-      $('button.edit-score').popover({
-        sanitize: false,
-        placement: 'auto',
-        container: 'body',
-        html: true,
-        content: function() {
-          var that = this
-          var assessmentInstanceId = $(this).data('ai-id')
-          var score = $(this).data('score')
-          var form = $('<form name="edit-total-score-perc-form" method="POST"></form>')
-          form.append('<input type="hidden" name="__action" value="edit_total_score_perc">')
-          form.append('<input type="hidden" name="__csrf_token" value="<%= __csrf_token %>">')
-          form.append('<input type="hidden" name="assessment_instance_id" value="' + assessmentInstanceId + '">')
-          var formGroup = $('<div class="form-group">')
-          var inputGroup = $('<div class="input-group">')
-          inputGroup.append('<input type="text" class="form-control" name="score_perc" value="' + score + '">')
-          inputGroup.append('<div class="input-group-append"><span class="input-group-text">%</span></div>')
-          formGroup.append(inputGroup)
-          form.append(formGroup)
-          form.append('<p><small>This change will be overwritten if further questions are answered by the student.</small></p>')
-          var cancel = $('<button type="button" class="btn btn-secondary mr-2">Cancel</button>')
-          cancel.on('click', function() {
-            $(that).popover('hide')
-          })
-          form.append(cancel)
-          form.append('<button type="submit" class="btn btn-primary">Change</button>')
-          return form
-        },
-        title: 'Change total percentage score',
-        trigger: 'click',
-      })
-    })
+        $('#gradebook-table').bootstrapTable({
+            onResetView: function () {
+                $('button.edit-score').popover({
```

Comment:
I'm completely fine to leave this as is, but this is one place where using ajax would be very nice to avoid the page reload. At the moment it's a bit annoying when you have carefully scrolled to a specific location the page and then the edit page reload scrolls back to the top 😄

Reply:
Ajax grade update implemented. The interesting thing about this particular update is that, for group assessments, the update needs to change more than one row, and the other rows may not be visible.

In any case, in order to get this to work properly with `bootstrap-table` I had to move the formatting of the grade fields to client side (since an update also requires reformatting), but I believe the current implementation makes this a bit more streamlined than before. This means that tests had to be changed.

Assuming the updated tests now pass, this should be ready for review.